eel/src/storage_persister.rs

```diff
@@ -263,34 +263,28 @@ impl StoragePersister {
             }
             StartupVariant::Recovery => {
                 // Try to get ChannelManager from remote
-                if self
+                let encrypted_data = self
                     .storage
-                    .object_exists(OBJECTS_BUCKET.to_string(), MANAGER_KEY.to_string())
+                    .get_object(OBJECTS_BUCKET.to_string(), MANAGER_KEY.to_string())
                     .map_to_runtime_error(
                         RuntimeErrorCode::RemoteStorageServiceUnavailable,
-                        "Failed to find a remote ChannelManager",
-                    )?
-                {
-                    let encrypted_data = self
-                        .storage
-                        .get_object(OBJECTS_BUCKET.to_string(), MANAGER_KEY.to_string())
-                        .map_to_runtime_error(
-                            RuntimeErrorCode::RemoteStorageServiceUnavailable,
-                            "Failed to read a remote ChannelManager",
-                        )?;
-                    let data = decrypt(&encrypted_data, &self.encryption_key)?;
-                    let mut buffer = Cursor::new(&data);
-                    let (block_hash, channel_manager) = <(
+                        "Failed to read a remote ChannelManager",
+                    )?;
+
+                if encrypted_data.is_empty() {
+                    Err(permanent_failure(
+                        "Failed to find remote ChannelManager even though this was determined to be a Recovery start",
+                    ))?;
+                }
```

Comment:
If there's no `ChannelManager` we won't get an empty vec, but an runtime error with `RuntimeErrorCode::ObjectNotFound`, right?

Reply:
Right, thanks! The idea was to do something like this: ba437dc866bfb70f862ebf34cd351ae5f9bc4c0f

Is there a way to make this more concise?

Reply:
Exactly what I was looking for, thank you :pray: